tools/disass/ida/disass.py

```diff
@@ -136,6 +139,87 @@ def get_direct_branch_target(branch_inst_ea):
         branch_inst_ea, target_ea))
     return target_ea
 
+# source: http://reverseengineering.stackexchange.com/a/1648/4197
```

Comment:
I'd like to see the below code to be removed for the following reasons:

 * The caching functionality is already covered by using things like `program.get_basic_block`. The `program` module also keeps track of more information, including subroutines, and block successors (including jump table targets).
 * This mechanism produces blocks that will be structured differently than Remill basic blocks. IDA's basic blocks do not end at `call` instructions.
 * This abstraction doesn't handle the case where two basic blocks overlap. Mapping an `ea` to a block is a one-to-many relationship. One place where this happens is glibc, where code will conditionally jump to a `cmpxchg` after a `lock` prefix. Another place that this can sometimes happen is in the Linux kernel.
 * The licensing of this code is unclear, despite it posted on a public StackExchange forum.